<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>653</width>
    <height>598</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>331</width>
      <height>531</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="2">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Сообщение</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLineEdit" name="lineEdit_M">
       <property name="inputMethodHints">
        <set>Qt::ImhNone</set>
       </property>
      </widget>
     </item>
     <item row="25" column="2">
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="5" column="2">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Моудль системы</string>
       </property>
      </widget>
     </item>
     <item row="6" column="2">
      <widget class="QLineEdit" name="lineEdit_p"/>
     </item>
     <item row="9" column="2">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Случайное число от 0 до p-1</string>
       </property>
      </widget>
     </item>
     <item row="10" column="2">
      <widget class="QLineEdit" name="lineEdit_k">
       <property name="enabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="17" column="2">
      <widget class="QLineEdit" name="lineEdit_r"/>
     </item>
     <item row="21" column="2">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Открытый ключ</string>
       </property>
      </widget>
     </item>
     <item row="22" column="2">
      <widget class="QLineEdit" name="lineEdit_y"/>
     </item>
     <item row="23" column="2">
      <widget class="QPushButton" name="pushButton_Verify">
       <property name="text">
        <string>Проверить ЭЦП</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QPushButton" name="pushButton_M">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>M</string>
       </property>
      </widget>
     </item>
     <item row="17" column="1">
      <widget class="QPushButton" name="pushButton_r">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>r</string>
       </property>
      </widget>
     </item>
     <item row="22" column="1">
      <widget class="QPushButton" name="pushButton_y">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>y</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QPushButton" name="pushButton_p">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>p</string>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <widget class="QPushButton" name="pushButton_k">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>k</string>
       </property>
      </widget>
     </item>
     <item row="15" column="2">
      <widget class="QPushButton" name="pushButton_Subscribe">
       <property name="text">
        <string>Выработать ЭЦП</string>
       </property>
      </widget>
     </item>
     <item row="16" column="2">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Первая половинка ЭЦП</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="formLayoutWidget">
    <property name="geometry">
     <rect>
      <x>339</x>
      <y>9</y>
      <width>301</width>
      <height>531</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
     </property>
     <item row="0" column="1">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Хэш сообщения</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="pushButton_m">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>m</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="lineEdit_m">
       <property name="enabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Число</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="pushButton_g">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>g</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="lineEdit_g">
       <property name="enabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Cекретный ключ</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QPushButton" name="pushButton_x">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>x</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLineEdit" name="lineEdit_x">
       <property name="enabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <layout class="QFormLayout" name="formLayout_2"/>
     </item>
     <item row="6" column="1">
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>25</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="7" column="1">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Вторая половинка ЭЦП</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QLineEdit" name="lineEdit_s"/>
     </item>
     <item row="8" column="0">
      <widget class="QPushButton" name="pushButton_s">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>s</string>
       </property>
      </widget>
     </item>
     <item row="14" column="1">
      <widget class="QLineEdit" name="lineEdit">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="13" column="1">
      <spacer name="verticalSpacer_3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="10" column="1">
      <widget class="QLineEdit" name="v1_2"/>
     </item>
     <item row="12" column="1">
      <widget class="QLineEdit" name="v2"/>
     </item>
     <item row="9" column="1">
      <widget class="QLabel" name="v1">
       <property name="text">
        <string>Результат V1</string>
       </property>
      </widget>
     </item>
     <item row="11" column="1">
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>Результат V2</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>653</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
